app/app_exception.php

```diff
@@ -18,3 +18,11 @@ class PageNotFoundException extends AppException
 class DuplicateEntryException extends AppException
 {
 }
+
+class PermissionException extends AppException
+{
+}
+
+class CategoryException extends AppException
```

Comment:
You may rename the following exceptions to a more specific one. Also consider adding PHPDoc :smile:
ex.
```php
/**
 * Exception thrown when invalid category is submitted
 */
class InvalidCategoryException extends AppException
{
}
```